soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -210,9 +222,9 @@ __weak void pm_state_set(enum pm_state state, uint8_t substate_id)
 		z_xt_ints_off(0xffffffff);
```

Comment:
Why do you need this here ? I'm looking these masks and I don't understand. This function enters will interruptions masked XCHAL_NMILEVEL. If you don't need to do it, you could move `core_desc[cpu].intenable = XTENSA_RSR("INTENABLE");` to `_save_core_context` and avoid unecessary instructions and duplicated code.

Reply:
For primary core _save_core_context is done only when CONFIG_ADSP_IMR_CONTEXT_SAVE is enabled.